src/dfcx_scrapi/core/test_cases.py

```diff
@@ -430,3 +432,149 @@ def calculate_coverage(self, coverage_type: int, agent_id: str = None):
         )
         response = client.calculate_coverage(request)
         return response
+
+    def _convert_flow(self, flow_id, flows_map):
+        """Gets a flow display name from a flow ID"""
+        if flow_id.split("/")[-1] == "-":
+            return ""
+        # flow_id_converted = str(agent_id) + '/flows/' + str(flow_id)
+        if flow_id in flows_map:
+            return flows_map[flow_id]
+        # TODO: Should throw error instead of returning default
+        return "Default Start Flow"
+
+    # Note that flow id includes agent, normally...
+    def _convert_page(self, page_id, flow_id, pages_map):
```

Comment:
@SeanScripts one way to optimize this section would be to create dictionary of your `page_id` mappings.

Then you can call `.get` on the dictionary directly and return `INVALID` if nothing matches.
This should significantly cut down on the number of if/else checks you need to do.

```py
page_dict = {
  'END_SESSION': 'End Session',
  'END_FLOW': 'End Flow',
  etc.
}

display_name = page_dict.get(page_id, 'INVALID')
```

Reply:
I added this dictionary and returned the mapping through it if the page ID was one of these special cases. However, we still need to look up the page name from the main dictionary of page names if it's not one of these special cases. I simplified the part after this a little bit by using `.get`.

Also, I know the latest version of Scrapi added the special cases to the list of pages, but I'm not sure what the format is, compared to what is returned in the test cases. In particular, whether or not the "pages" like END_SESSION include the flow ID. I might need to check that.

Reply:
Looks like with the 1.6 update to `get_pages_map`, this check for the special pages isn't needed at all. Thanks! This function has been simplified.